x-pack/plugins/cross_cluster_replication/public/app/components/follower_index_form.js

```diff
@@ -92,29 +90,60 @@ export const FollowerIndexForm = injectI18n(
       const followerIndex = isNew
         ? getEmptyFollowerIndex(this.props.remoteClusters)
         : {
+          ...getEmptyFollowerIndex(),
           ...this.props.followerIndex,
         };
 
       this.state = {
+        isNew,
         followerIndex,
-        fieldsErrors: validateFollowerIndex(followerIndex),
+        fieldsErrors: {},
         areErrorsVisible: false,
-        isNew,
+        areAdvancedSettingsVisible: false,
       };
 
       this.validateIndexName = debounce(this.validateIndexName, 500);
     }
 
     onFieldsChange = (fields) => {
-      const errors = validateFollowerIndex(fields);
       this.setState(updateFields(fields));
-      this.setState(updateFormErrors(errors));
 
       if (this.props.apiError) {
         this.props.clearApiError();
       }
     };
 
+    onFieldsErrorChange = (errors) => {
+      this.setState(updateFormErrors(errors));
+    }
+
+    onIndexNameChange = ({ name }) => {
+      this.onFieldsChange({ name });
+      this.validateIndexName(name);
+    }
+
+    validateIndexName = async (name) => {
+      if (!name || !name.trim) {
+        return;
+      }
+
+      const { intl } = this.props;
+
+      try {
+        const indices = await loadIndices();
+        const doesExist = indices.some(index => index.name === name);
+        if (doesExist) {
+          const message = intl.formatMessage({
+            id: 'xpack.crossClusterReplication.followerIndexForm.indexAlreadyExistError',
+            defaultMessage: 'An index with the same name already exists.'
```

Comment:
since we are only checking follower indices, maybe we should be explicit about that and say `A follower index with the same name already exists.`? I imagine that if the user inputs an index name that matches a non-follower index, we will bubble up an ES error

Reply:
We are actually checking all the user indices and prevent a follower name to equal an existing index name. Even before ES throws an error 😊